src/overlays/actors/ovl_Door_Shutter/z_door_shutter.c

```diff
@@ -612,25 +805,26 @@ void func_809975C0(DoorShutter* this, PlayState* play) {
     }
 }
 
-void func_809976B8(DoorShutter* this, PlayState* play) {
-    f32 mult;
+void DoorShutter_GohmaBlock(DoorShutter* this, PlayState* play) {
+    f32 bounceFactor;
 
+    // Bounce a bit (unnoticeable in-game)
     if (this->unk_164 != 0) {
         this->unk_164--;
-        mult = sinf(this->unk_164 * 250.0f / 100.0f);
-        this->dyna.actor.shape.yOffset = this->unk_164 * 3.0f / 10.0f * mult;
+        bounceFactor = sinf(this->unk_164 * 250.0f / 100.0f);
+        this->dyna.actor.shape.yOffset = this->unk_164 * 3.0f / 10.0f * bounceFactor;
     }
 }
 
-void func_80997744(DoorShutter* this, PlayState* play) {
-    f32 phi_f0;
+void DoorShutter_PhantomGanonBars(DoorShutter* this, PlayState* play) {
+    f32 targetOffsetY;
```

Comment:
any way we can make these more of an action name? the gohma block one is mostly just idle except for the little bouncing.
And the bars one is actually confusing. Theres no logic to make sure link is in the arena before counting down and closing. Does phantom ganon spawn the bars only after you get to the center of the arena? lol

Reply:
> any way we can make these more of an action name?

`DoorShutter_BeGohmaBlock` and `DoorShutter_BePhantomGanonBars` 8) ?

DoorShutter is a complicated actor, I'd rather avoid making things look more generic than they are but idk (like, not `DoorShutter_Bounce` and `DoorShutter_Raise`)

These actions also serve as the idle action for those types, but maybe it'd still be fine to do `DoorShutter_GohmaBlockBounce` (though it's invisible) and `DoorShutter_PhantomGanonBarsRaise` idk

> Does phantom ganon spawn the bars only after you get to the center of the arena?

yes, and so does Queen Gohma 8)

Reply:
yea its tricky that it does both the closing and the idle.
I would almost prefer it be "update gohma block" instead of just "gohma block", but idk. more opinions welcome

Reply:
note `DoorShutter_GohmaBlockFalling` is a thing

Reply:
now all infinitives besides `DoorShutter_UpdateOpening`